pennylane/math/multi_dispatch.py

```diff
@@ -798,6 +798,13 @@ def iscomplex(tensor, like=None):
     return np.iscomplex(tensor)
 
 
+@multi_dispatch()
+def size(tensor, like=None):
+    """Return the size of the tensor."""
+    s = tensor.numel() if like == "torch" else np.size(tensor)
+    return np.array(s, like=like)

```

Comment:
Why are we converting to an array at the end? Shouldn't this just be an integer?

Reply:
To avoid losing the interface information. If we don't do this:
```pycon
>>> import torch
>>> import pennylane as qml
>>> a = torch.tensor([1, 2, 3])
>>> qml.math.get_interface(qml.math.size(a))
'numpy'
```

Reply:
Also, some tests were failing when returning an integer. I thought this was the problem (or maybe differentiability?). Let me figure it out!

Reply:
Nvm, `size` returns an int now